docs/porting-guide.rst

```diff
@@ -138,73 +138,265 @@ In particular, when running on top of CPython, HPy will automatically provide
+    ``PyBytes_FromString``                           :c:func:`HPyBytes_FromString`
+    ``PyUnicode_FromString``                         :c:func:`HPyUnicode_FromString`
+    ``PyUnicode_Check``                              :c:func:`HPyUnicode_Check`
+    ``PyUnicode_AsASCIIString``                      :c:func:`HPyUnicode_AsASCIIString`
+    ``PyUnicode_AsLatin1String``                     :c:func:`HPyUnicode_AsLatin1String`
+    ``PyUnicode_AsUTF8String``                       :c:func:`HPyUnicode_AsUTF8String`
+    ``PyUnicode_AsUTF8AndSize``                      :c:func:`HPyUnicode_AsUTF8AndSize`
+    ``PyUnicode_FromWideChar``                       :c:func:`HPyUnicode_FromWideChar`
+    ``PyUnicode_DecodeFSDefault``                    :c:func:`HPyUnicode_DecodeFSDefault`
+    ``PyUnicode_DecodeFSDefaultAndSize``             :c:func:`HPyUnicode_DecodeFSDefaultAndSize`
+    ``PyUnicode_EncodeFSDefault``                    :c:func:`HPyUnicode_EncodeFSDefault`
+    ``PyUnicode_ReadChar``                           :c:func:`HPyUnicode_ReadChar`
+    ``PyUnicode_DecodeASCII``                        :c:func:`HPyUnicode_DecodeASCII`
+    ``PyUnicode_DecodeLatin1``                       :c:func:`HPyUnicode_DecodeLatin1`
+    ``PyList_Check``                                 :c:func:`HPyList_Check`
+    ``PyList_New``                                   :c:func:`HPyList_New`
+    ``PyList_Append``                                :c:func:`HPyList_Append`
+    ``PyDict_Check``                                 :c:func:`HPyDict_Check`
+    ``PyDict_New``                                   :c:func:`HPyDict_New`
+    ``PyTuple_Check``                                :c:func:`HPyTuple_Check`
+    ``PyImport_ImportModule``                        :c:func:`HPyImport_ImportModule`
+    ``PyCapsule_IsValid``                            :c:func:`HPyCapsule_IsValid`
+    ``PyEval_RestoreThread``                         :c:func:`HPy_ReenterPythonExecution`
+    ``PyEval_SaveThread``                            :c:func:`HPy_LeavePythonExecution`
+    ================================================ ================================================
+
+
+.. note: There are, of course, also cases where it is not possible to map directly and safely from a C API function (or concept) to an HPy API function (or concept).
+
+Reference Counting ``Py_INCREF`` and ``Py_DECREF``
+--------------------------------------------------
+
+The equivalents of ``Py_INCREF`` and ``Py_DECREF`` are essentially
+:c:func:`HPy_Dup` and :c:func:`HPy_Close`, respectively. The main difference is
+that :c:func:`HPy_Dup` gives you a *new handle* to the same object which means
+that the two handles may be different if comparing them with ``memcmp`` but
+still reference the same object. As a consequence, you may close a handle only
+once, i.e., you cannot call :c:func:`HPy_Close` twice on the same ``HPy``
+handle. For examples, see also sections :ref:`api:handles` and :ref:`api:handles
```

Comment:
In general, terms should only be converted to links on first use. Subsequent use should get regular formatting above.

Maybe this would help make the point about closing handles more clearly?
```suggestion
handle, even if returned from ``HPy_Dup``. For examples, see also sections :ref:`api:handles` and :ref:`api:handles
```